### PR TITLE
add label colors

### PR DESCRIPTION
**Pull request by rebornix**
Add label colors. Since GitHub API doesn't tell us about the right foreground color to use, we do the math ourselves.

### CODE REVIEW

**Comment by GeorgeWL** [comment]
I think that colour shouldn't be a required field though. Colour should be a `color?:string` interface.

That way if someone has no interest in colouring the labels, they can keep on using the current implementation.

File: src/github/interface.ts [hunk 1]
```diff
@@ -56,6 +56,7 @@ export interface IGitHubRef {
 
 export interface ILabel {
 	name: string;
+	color: string;
```
Comment (GeorgeWL) [inline src/github/interface.ts]: I think that colour shouldn't be a required field though. Colour should be a color?:string interface.

That way if someone has no interest in colouring the labels, they can keep on using the current implementation.

File: src/github/pullRequestModel.ts [hunk 2]
```diff
@@ -75,7 +75,7 @@ export class PullRequestModel {
 		this.bodyHTML = prItem.bodyHTML;
 		this.html_url = prItem.url;
 		this.author = prItem.user;
-		this.labels = (prItem.labels || []).map(label => label.name);
+		this.labels = (prItem.labels || []);
```
Comment (GeorgeWL) [inline src/github/pullRequestModel.ts]: what if they've only set name and not the color?

**Comment by alexr00** [comment]
In the time since this PR was opened we've added label colors.

**Pull request closed without merging**